src/main/java/io/reactivex/internal/schedulers/RxThreadFactory.java

```diff
@@ -30,10 +30,7 @@ public RxThreadFactory(String prefix) {
     
     @Override
     public Thread newThread(Runnable r) {
-        StringBuilder nameBuilder = new StringBuilder();
-        nameBuilder.append(prefix)
-        .append('-')
-        .append(incrementAndGet());
+        StringBuilder nameBuilder = new StringBuilder(prefix).append('-').append(incrementAndGet());
```

Comment:
Using the constructor initializes the underlying `char[]` to length + 16 instead of the default constructors size of 16 which almost always had to do an expansion to fit the prefix.